common/static/coffee/src/discussion/views/discussion_topic_menu_view.js

```diff
@@ -187,6 +189,38 @@
                     }
                 }
                 return name;
+            },
+
+            // TODO: this helper class duplicates functionality in DiscussionThreadListView.filterTopics
+            // for use with a very similar category dropdown in the New Post form.  The two menus' implementations
+            // should be merged into a single reusable view.
+            filterDrop: function (e) {
+                var $drop, $items, query;
+                $drop = $(e.target).parents('.topic-menu-wrapper');
+                query = $(e.target).val();
+                $items = $drop.find('.topic-menu-item');
+
+                if (query.length === 0) {
+                    $items.removeClass('hidden');
+                    return;
+                }
+
+                $items.addClass('hidden');
+                $items.each(function () {
+                    var path, pathText, pathTitles;
+                    path = $(this).parents(".topic-menu-item").andSelf();
```

Comment:
I prefer using the element passed to the callback function rather than `this` because it's much more explicit and doesn't rely on working knowledge of `jQuery.each`.

For example:
```javascript
$items.each(function (_index, item) {
    var path, pathText, pathTitles;
    path = $(item).parents(".topic-menu-item").andSelf();
    ...
});
```